op-e2e/actions/upgrades/interop_fork_test.go

```diff
@@ -0,0 +1,138 @@
+package upgrades
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-e2e/actions/helpers"
+	"github.com/ethereum-optimism/optimism/op-e2e/actions/interop/dsl"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-service/predeploys"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/lmittmann/w3"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInteropUpgradeTransactions(gt *testing.T) {
+	t := helpers.NewDefaultTesting(gt)
+
+	// Run Isthmus until the interop upgrade
+	is := dsl.SetupInterop(t, dsl.SetInteropOffset(uint64(15)))
+	actors := is.CreateActors()
+	actors.PrepareChainState(t)
+
+	////////////////////////////
+	// Pre-upgrade Block Production
+	////////////////////////////
+
+	fmt.Println("Pre-upgrade Block Production")
+
+	// Start op-nodes
+	actors.ChainA.Sequencer.ActL2PipelineFull(t)
+	actors.ChainB.Sequencer.ActL2PipelineFull(t)
+
+	// Verify Interop is not active at genesis yet
+	l2Head := actors.ChainA.Sequencer.L2Unsafe()
+	require.NotZero(t, l2Head.Hash)
+	require.True(t, is.Out.L2s[actors.ChainA.ChainID.String()].RollupCfg.IsIsthmus(l2Head.Time), "Isthmus should be active at genesis in chain A")
+	require.False(t, is.Out.L2s[actors.ChainA.ChainID.String()].RollupCfg.IsInterop(l2Head.Time), "Interop should not be active at genesis in chain A")
+
+	l2Head = actors.ChainB.Sequencer.L2Unsafe()
+	require.NotZero(t, l2Head.Hash)
+	require.True(t, is.Out.L2s[actors.ChainB.ChainID.String()].RollupCfg.IsIsthmus(l2Head.Time), "Isthmus should be active at genesis in chain B")
+	require.False(t, is.Out.L2s[actors.ChainB.ChainID.String()].RollupCfg.IsInterop(l2Head.Time), "Interop should not be active at genesis in chain B")
+
+	// Submit all and mine for both chains
+	actors.ChainA.Sequencer.ActL2EmptyBlock(t)
+	actors.ChainA.Batcher.ActSubmitAll(t)
+	actors.ChainB.Sequencer.ActL2EmptyBlock(t)
+	actors.ChainB.Batcher.ActSubmitAll(t)
+	batchTxA := actors.ChainA.Batcher.LastSubmitted
+	batchTxB := actors.ChainB.Batcher.LastSubmitted
+	actors.L1Miner.ActL1StartBlock(12)(t)
+	actors.L1Miner.ActL1IncludeTxByHash(batchTxA.Hash())(t)
+	actors.L1Miner.ActL1IncludeTxByHash(batchTxB.Hash())(t)
+	actors.L1Miner.ActL1EndBlock(t)
```

Comment:
Except for the `actors.ChainX.Sequencer.ActL2EmptyBlock(t)` calls, all this could be lumped into a helper method on `actors` like `ActBatchAllAndMine`, that batch submits for all chains and mines all batcher txs, wdyt? We have the same helper on the action tests env.

Reply:
But note that the result of this is currently not used in this test. No derivation is happening at all.

It would be better to move `ActBatchAllAndMine` to the end of the test, then do derivation using
```
	actors.ChainA.Sequencer.ActL2PipelineFull(t)
	actors.ChainB.Sequencer.ActL2PipelineFull(t)
```
(which performs the actual derivation) afterwards, then assert that the so derived (local) safe chain heads are the same as the unsafe heads.

Reply:
Created a helper function `ActBatchAllAndMine()` and moved to the end of the test, where I added assertions that show the derivation proceeds successfully after upgrade.